### PR TITLE
refactor: prevent panicking with params

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Use `GetParamsIfExist` to prevent panic

Closes: https://github.com/zeta-chain/node/issues/1860

### CODE REVIEW

File: x/emissions/keeper/params.go [hunk 1]
```diff
@@ -5,13 +5,14 @@ import (
 	"github.com/zeta-chain/zetacore/x/emissions/types"
 )
 
-// GetParams get all parameters as types.Params
-func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
-	k.paramstore.GetParamSet(ctx, &params)
+// GetParamSetIfExists get all parameters as types.Params if they exist
+// non existent parameters will return zero values
+func (k Keeper) GetParamSetIfExists(ctx sdk.Context) (params types.Params) {
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/emissions/keeper/params.go:10]: Let's rename to `GetParamsIfExists` everywhere

File: x/emissions/types/distributions.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetDistributions returns the current distribution of rewards
+// for validators, observers and TSS signers
+// If the percentage is not set, it returns 0
+func GetDistributions(params Params) (sdkmath.Int, sdkmath.Int, sdkmath.Int) {
+	// Fetch the validator rewards, use 0 if the percentage is not set
+	validatorRewards := sdk.NewInt(0)
+	validatorRewardsDec, err := sdk.NewDecFromStr(params.ValidatorEmissionPercentage)
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/emissions/types/distributions.go:14]: This method is no longer used and must be removed https://github.com/zeta-chain/node/issues/1813

File: x/emissions/types/distributions.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetDistributions returns the current distribution of rewards
+// for validators, observers and TSS signers
+// If the percentage is not set, it returns 0
+func GetDistributions(params Params) (sdkmath.Int, sdkmath.Int, sdkmath.Int) {
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/emissions/types/distributions.go]: 0038a2a7

File: x/emissions/keeper/block_rewards_components.go [hunk 4]
```diff
@@ -17,9 +17,9 @@ func (k Keeper) GetBlockRewardComponents(ctx sdk.Context) (sdk.Dec, sdk.Dec, sdk
 	return reservesFactor, bondFactor, durationFactor
 }
 func (k Keeper) GetBondFactor(ctx sdk.Context, stakingKeeper types.StakingKeeper) sdk.Dec {
-	targetBondRatio := sdk.MustNewDecFromStr(k.GetParams(ctx).TargetBondRatio)
-	maxBondFactor := sdk.MustNewDecFromStr(k.GetParams(ctx).MaxBondFactor)
-	minBondFactor := sdk.MustNewDecFromStr(k.GetParams(ctx).MinBondFactor)
+	targetBondRatio := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).TargetBondRatio)
+	maxBondFactor := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).MaxBondFactor)
+	minBondFactor := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).MinBondFactor)
```
Comment (kingpinXD) [inline x/emissions/keeper/block_rewards_components.go:22]: Why dont we just remove these unused params.
https://github.com/zeta-chain/node/issues/1852

File: x/emissions/keeper/block_rewards_components.go [hunk 5]
```diff
@@ -37,10 +37,10 @@ func (k Keeper) GetBondFactor(ctx sdk.Context, stakingKeeper types.StakingKeeper
 }
 
 func (k Keeper) GetDurationFactor(ctx sdk.Context) sdk.Dec {
-	avgBlockTime := sdk.MustNewDecFromStr(k.GetParams(ctx).AvgBlockTime)
+	avgBlockTime := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).AvgBlockTime)
```
Comment (kingpinXD) [inline x/emissions/keeper/block_rewards_components.go:40]: same comment as above

File: x/emissions/keeper/block_rewards_components.go (hunk 4 above)
Comment (lumtis) [reply x/emissions/keeper/block_rewards_components.go:22]: I can remove this in this PR but overall we should address the whole investigation on what must be removed in this one: https://github.com/zeta-chain/node/issues/1813

Comment (lumtis) [reply x/emissions/keeper/block_rewards_components.go:22]: Looked at the code. I think it is better that we address all removal in the same PR.

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**